src/Resolvers/TransformedDataCollectableResolver.php

```diff
@@ -82,7 +82,7 @@ protected function transformPaginator(
         Wrap $wrap,
         TransformationContext $nestedContext,
     ): array {
-        $paginator = $paginator->through(fn (BaseData $data) => $this->transformationClosure($nestedContext)($data));
+        $items = array_map(fn (BaseData $data) => $this->transformationClosure($nestedContext)($data), $paginator->items());
 
         if ($nestedContext->transformValues === false) {
             return $paginator->all();
```

Comment:
I think somehow, we also should be able to somehow run the items here through the closure? If you pull in the latest changes of this file you'll see that the `transformationClosure` is a bit more complicated than it was. Maybe we can use through here? Since we're not transforming values?

Reply:
I'm sorry, I'm not sure what you mean?

Reply:
The items in:

```php
 return $paginator->all();
```

Will now not run through the `transformationClosure`, which may cause problems since the new version of `transformationClosure` also performs operations on the data objects even if we're not transforming values.

Reply:
Okay, could that be done in another PR? If you'd rather have it in this PR, I'm not sure what the behavior should be. Remove the guard clause in `transformPaginator`? Or adapt the early return to return the mapped items?

Reply:
I think calling this should be enough:

```php
 return $paginator->through(fn (BaseData $data) => $this->transformationClosure($nestedContext)($data))->all();
```